cfme/base/login.py

```diff
@@ -30,6 +29,29 @@ def logged_in_as_user(self, user):
 
         return user.name == self.current_fullname
 
+    def change_group(self, group_name):
+        """ From the settings menu change to the group specified by 'group_name'
+            Only available in versions >= 5.9
+
+            User is required to be currently logged in
+        """
+        if self.extra.appliance.version < '5.9':
+            raise CFMEException("Changing the user group is not supported in versions < 5.8")
+
+        if not self.logged_in_as_user:
+            raise CFMEException("Unable to change group when a user is not logged in")
+
+        if group_name not in self.group_names:
+            raise CFMEException("{} is not an assigned group for {}".format(
+                group_name, self.current_username))
+
+        if self.current_groupname == group_name:
```

Comment:
**QUESTION:** Normally in just a library to enable someone to do something, and not sitting in a test framework, I would do exactly this.   However I wonder if there is a situation where you want to test setting the group to the same group you are in, to make sure no awfulness happens when that is done?

Reply:
That's a valid point. The current_groupname check was added for ease of use so the caller can blindly set the group w/o concern that it's already active since trying to change to the currently assigned group will throw a WidgetOperationFailed exception.